fix error when it is not in json format

YingXue opened the pull request:
When response is not json, it throws error that looks like bug in the code: an example here: https://github.com/Azure/azure-iot-explorer/issues/138

Added a try catch with a bit of generic hep text
![image](https://user-images.githubusercontent.com/5489222/67808583-2ff08700-fa54-11e9-9690-cce8e03d831f.png)

Some tests start to fail possibly due to the deprecation warnings start to become errors. Fixed in second commit.

andrew-buckley commented:
Is it always because the connection string is wrong?

YingXue commented:
> Is it always because the connection string is wrong? The error message seems to be more specific than generic.

Agree... I changed the wording to be slightly loose. Most of time, if the response from the API doesn't contain any json body (which is the case we are seeing error with no messages) are the case we will show this 'generic error', which mostly due to invalid connection string.

andrew-buckley commented:
> > Is it always because the connection string is wrong? The error message seems to be more specific than generic.
> 
> Agree... I changed the wording to be slightly loose. Most of time, if the response from the API doesn't contain any json body (which is the case we are seeing error with no messages) here, are the case we will show this 'generic error', which mostly due to invalid connection string.

Ideally we would have a regex check on the connection string and report specifically what is invalid about the connection string. Otherwise without a response from the server we don't *really* know what's going on.

YingXue commented:
> > > Is it always because the connection string is wrong? The error message seems to be more specific than generic.
> > 
> > 
> > Agree... I changed the wording to be slightly loose. Most of time, if the response from the API doesn't contain any json body (which is the case we are seeing error with no messages) here, are the case we will show this 'generic error', which mostly due to invalid connection string.
> 
> Ideally we would have a regex check on the connection string and report specifically what is invalid about the connection string. Otherwise without a response from the server we don't _really_ know what's going on.

I've updated the helper to check if it contains the repo id. But even with the regex check, we don't know if it is wrong or not. A example is, if a connection string is deleted from the portal, then user would not get any reponse back (which belongs to the cases I am handling here).

andrew-buckley commented:
> > > > Is it always because the connection string is wrong? The error message seems to be more specific than generic.
> > > 
> > > 
> > > Agree... I changed the wording to be slightly loose. Most of time, if the response from the API doesn't contain any json body (which is the case we are seeing error with no messages) here, are the case we will show this 'generic error', which mostly due to invalid connection string.
> > 
> > 
> > Ideally we would have a regex check on the connection string and report specifically what is invalid about the connection string. Otherwise without a response from the server we don't _really_ know what's going on.
> 
> I've updated the helper to check if it contains the repo id. But even with the regex check, we don't know if it is wrong or not. A example is, if a connection string is deleted from the portal, then user would not get any reponse back (which belongs to the cases I am handling here).
> 
> Also, those cases, they won't even fail at the api step, as it would fail the connection string text field validation.
> It is the cases that it cannot be pre-validated, and still the api has no response body, is where we want to show users of this error.

I guess what I'm saying is we only really know as much as what the API response is telling us. For the same reason you referenced above, we can't distinguish between the connection string being wrong, the connection string keys being removed, the hub being deleted, or any number of scenarios. So what I'm saying is it seems like we're trying to infer something specific from no information since the response body is empty which is think could be problematic. The information we do have is the connection string and the regex which we can easily validate. Which is why I prefer the new wording because if it turns out not to be a connection string problem we weren't determining with certainty that this was the issue.
